### PR TITLE
Add steps to migrate from v2 to v3

### PR DESCRIPTION
**Pull request by xino12**
#### Description of the changes
Add steps to ease the migration from `v2` to `v3` 

### Author

- [x] add a risk label after carefully considering the "blast radius" of your changes
- [x] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [x] assign at least one reviewer

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: docs/v2tov3.md [hunk 1]
```diff
@@ -147,6 +147,71 @@ For a complete view of the GoSDK `v3`, please refer to [TODO: link API and tutor
...
+integration src folder :
+
+For **Linux**:
+```bash
+find . -type f -iname "*.go" |xargs -i {} sed -i '' 's/github.com/newrelic/infra-integrations-sdk/sdk/github.com/newrelic/infra-integrations-sdk/integration/' {}
+find . -type f -iname "*.go" |xargs -i {} sed -i '' 's/github.com/newrelic/infra-integrations-sdk/cache/github.com/newrelic/infra-integrations-sdk/persist/' {}
+find . -type f -iname "*.go" |xargs -i {} sed -i '' 's/github.com/newrelic/infra-integrations-sdk/metric/github.com/newrelic/infra-integrations-sdk/data/metric/' {}
+``` 
+
+For **MacOS**:
```
Comment (varas) [inline docs/v2tov3.md]: aren't the same as Linux?

Comment (xino12) [reply docs/v2tov3.md]: I think that by default the `xargs` command in MacOS does not have the -i option and is replaced by -I.

And in linux -i works by default.

Comment (xino12) [reply docs/v2tov3.md]: Oh, I see that the -i option has been deprecated. I'll change that.
http://man7.org/linux/man-pages/man1/xargs.1.html

File: docs/v2tov3.md [hunk 2]
```diff
@@ -147,6 +147,63 @@ For a complete view of the GoSDK `v3`, please refer to [TODO: link API and tutor
...
+storer, _ := persist.NewFileStore(storagePath string, ilog log.Logger, ttl time.Duration)
+populatedMetrics, _ := metric.NewSet("NewSet", storer)
+```
+
+Besides all of the changes mentioned above these are some terms that you can look for in your integration code and the `v3`
+ version of them.
+
+| v2                      | v3                              |
+| ----------------------- | ------------------------------- |
+| sdk.NewIntegration()    | integration.New()               |
```
Comment (varas) [inline docs/v2tov3.md]: 🏆

File: docs/v2tov3.md [hunk 3]
```diff
@@ -147,6 +147,63 @@ For a complete view of the GoSDK `v3`, please refer to [TODO: link API and tutor
...
+ ```bash
+ integration, err := integration.New(integrationName, integrationVersion)
+ e := integration.Entity("Entity1", "test") # Local or remote entity it depends in the use case
+ e.NewMetricSet("TestSample")
+ ```
+
+When creating a new Metric Set a storer object is needed in order for the integration to be able to persist them into disk.
+
+```bash
+storer, _ := persist.NewFileStore(storagePath string, ilog log.Logger, ttl time.Duration)
```
Comment (areina) [inline docs/v2tov3.md]: Do we want to document this? In theory, people would create metric sets using the `NewMetricSet` method and not this one, right?

File: docs/v2tov3.md [hunk 4]
```diff
@@ -147,6 +147,63 @@ For a complete view of the GoSDK `v3`, please refer to [TODO: link API and tutor
 * Some member names changed, e.g. `MetricSet` is now `metric.Set`
 * Added `EntityMetadata` struct to fulfill the [JSON `v2` schema](#json-v2-schema-changes).
 
+## Migrate from `v2` to `v3` steps
+### Imports
+* First of all we've renamed some packages. So the first you can do when migrating from `v2` to `v3` is run these commands inside your 
+integration src folder :
+
+```bash
+find . -type f -iname "*.go" |xargs -I {} sed -i '' 's/github.com/newrelic/infra-integrations-sdk/sdk/github.com/newrelic/infra-integrations-sdk/integration/' {}
```
Comment (areina) [inline docs/v2tov3.md]: Just a minor about the sed statement. With sed, you can use any separator char, `/` is the standard but in some cases like this one I feel it is very confusing. 
You could replace it by: 
`sed -i '' 's#github.com/newrelic/infra-integrations-sdk/sdk#github.com/newrelic/infra-integrations-sdk/integration#'`

File: docs/v2tov3.md (hunk 3 above)
Comment (xino12) [reply docs/v2tov3.md]: So true.

File: docs/v2tov3.md (hunk 4 above)
Comment (xino12) [reply docs/v2tov3.md]: 👍

File: docs/v2tov3.md [hunk 5]
```diff
@@ -147,6 +147,56 @@ For a complete view of the GoSDK `v3`, please refer to [TODO: link API and tutor
...
+ e.NewMetricSet("TestSample")
+ ```
+
+Besides all of the changes mentioned above these are some terms that you can look for in your integration code and the `v3`
+ version of them.
+
+| v2                      | v3                              |
+| ----------------------- | ------------------------------- |
+| sdk.NewIntegration()    | integration.New()               |
+| args.All                | args.All()                      |
```
Comment (mariomac) [inline docs/v2tov3.md]: Just to be picky: keep these entries in `code fixed-length font`.

For the rest, it looks good for me. It has my approval if @areina is satisfied with the way you addressed his comments.

**Pull request merged**